### ISSUE TITLE
Allow/Enforce the same password and expiration requirements on Public Album links

### ISSUE THREAD

**Issue by ACiDGRiM** [nextcloud/photos#1568]
**Is your feature request related to a problem? Please describe.**
I am totally in love with the Albums feature, this is what I have been waiting for a very long time. One issue I have is that there is not an option to password protect and set expiration dates on public Album links.

**Describe the solution you'd like**
I would like to be able to create multiple links,with different passwords/no password/expiration dates, just like I can for sharing files and folders.

**Describe alternatives you've considered**
This is a new feature, so I don't believe there are any alternatives, other than the existing option to share a folder with copies of media. However this loses the benefit of not having to create multiple copies of the same picture for different folders, whereas the Albums can share the same picture file across multiple collections.

**Additional context**
The only option for the public link available is to disable and create a new link.

**Comment by marcopenhacking** [comment]
For default links (non album links) it's also possible to disable download buttons. It would be good, if public album links would support this feature as well. See also: [https://nextcloud.com/de/blog/secure-view-prevent-your-shared-files-from-getting-downloaded/]()

**Comment by IeP4nieF** [comment]
I only can repeat @relink2013 : "I just spent the last several hours digging through all my settings and searching online trying to figure this out. The feature is so obvious I assumed something was configured wrong or it was a bug."

The sharing-function is useless without the ability of setting up a password-function. And of course there must also be the possibility to edit the sharing, like delete the share, setup an vanish-date, grant rights and so on.

**Comment by CAberry** [comment]
I must bump this feature request. 

Obviously, sharing an album of an event with family and friends without being able to password protect it is simply a no go.  We can not create accounts for every one just for a one time share.

Greetings,

**Comment by darguez** [comment]
I agree. Password protect in shared public links for albums should be a must nowadays (in the same way it is for folders). I love the posibility of sharing albums with family (just with selected photos), but I don't do it because I fear the link may accidentally fall into the wrong hands.

Is this feature in the roadmap? It would be amazing to have it.

**Comment by darguez** [comment]
Curious thing is the workaround of placing this code in nextcloud .htaccess to force Apache2 to ask for a password when accessing shared albums links returns a nextcloud Error page (401) after entering valid credentials.
```
<If "%{REQUEST_URI} =~ m#^/index.php/apps/photos/public/#">
    AuthUserFile /etc/apache2/.htpasswd
    AuthName "Password protected"
    AuthType Basic
    Require valid-user
</If>
```
**Important:**

To generate a user/pass you I used `sudo htpasswd -c /etc/apache2/.htpasswd username`

And I had to set `AllowOverrride` to `All` instead `None` in `/etc/apache2/apache2.conf`
```
<Directory /var/www/>
        Options Indexes FollowSymLinks
        AllowOverride All
        Require all granted
</Directory>
```
Apache2 restart is also required after conf changes: `sudo systemctl restart apache2.service`

**Comment by darguez** [comment]
> Curious thing is the workaround of placing this code in nextcloud .htaccess to force Apache2 to ask for a password when accessing shared albums links returns a nextcloud Error page (401) after entering valid credentials.
> 
> ```
> <If "%{REQUEST_URI} =~ m#^/index.php/apps/photos/public/#">
>     AuthUserFile /etc/apache2/.htpasswd
>     AuthName "Password protected"
>     AuthType Basic
>     Require valid-user
> </If>
> ```
> 
> **Important:**
> 
> To generate a user/pass you I used `sudo htpasswd -c /etc/apache2/.htpasswd username`
> 
> And I had to set `AllowOverrride` to `All` instead `None` in `/etc/apache2/apache2.conf`
> 
> ```
> <Directory /var/www/>
>         Options Indexes FollowSymLinks
>         AllowOverride All
>         Require all granted
> </Directory>
> ```
> 
> Apache2 restart is also required after conf changes: `sudo systemctl restart apache2.service`

There is a workaround to avoid Nextcloud 401 error when using this .htaccess patch, in `config/config.php` add at the beginning:
```
<?php
// PATCH BEGIN
// If URL includes a shared album...
if (isset($_SERVER['REQUEST_URI']) && strpos($_SERVER['REQUEST_URI'], '/apps/photos/public/') !== false) {
    // ...manually delete authentication traces to keep the anonymous access
    unset($_SERVER['PHP_AUTH_USER']);
    unset($_SERVER['PHP_AUTH_PW']);
    unset($_SERVER['HTTP_AUTHORIZATION']);
    unset($_SERVER['REDIRECT_HTTP_AUTHORIZATION']);
}
// PATCH END
 
$CONFIG = array(
// ... here the existing configuration ...
``` 

If you want to protect memories albums, you can change the path to: `/apps/memories/a/`
